docker/daemon/client.go

```diff
@@ -36,7 +36,9 @@ func newDockerClient(sys *types.SystemContext) (*dockerclient.Client, error) {
 	}
 	var httpClient *http.Client
 	if url.Scheme != "unix" {
-		if url.Scheme == "http" {
+		if sys.HTTPClient != nil {
```

Comment:
`sys == nil` is always valid, and must be treated the same as `&types.SystemContext{}`.